pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -924,10 +939,13 @@ func (vs *VSphere) DiskIsAttached(volPath string, nodeName k8stypes.NodeName) (b
 	}
 
 	if !nodeExist {
-		glog.Warningf(
-			"Node %q does not exist. DiskIsAttached will assume vmdk %q is not attached to it.",
+		glog.Warningf("Node %q does not exist. DisksAreAttached will delete any vmdk %v attached to it.",
 			vSphereInstance,
 			volPath)
+		if err = vs.DetachDisk(volPath, nodeName); err != nil {
```

Comment:
Actually since this function is only for checking the status, maybe we should put detachdisk function inside of it. When node node no longer exists, we should return an error message instead of returning false indicating volume is already detached.